Documentation/op-guide/performance.md

```diff
@@ -10,4 +10,40 @@ There are other sub-systems which impact the overall performance of etcd. Each s
 
 ## Benchmarks
 
-TODO
\ No newline at end of file
+To benchmark etcd performance, use [benchmark](https://github.com/coreos/etcd/tree/master/tools/benchmark) CLI tool. Following results are from a 3 member etcd cluster with the hardware configuration:
+
+- Google Cloud Compute Engine
+- 3 machines of 8 vCPUs + 16GB Memory + 50GB SSD
+- 1 machine(client) of 16 vCPUs + 30GB Memory + 50GB SSD
+- Ubuntu 15.10
+- etcd v3 master branch (commit SHA d8f325d), Go 1.6.2
+
+These are rough numbers. etcd can write:
+
+| Number of keys | Key size in bytes | Value size in bytes | Number of connections | Number of clients | Target etcd server | Average write QPS | Memory |
+|----------------|-------------------|---------------------|-----------------------|-------------------|--------------------|-------------------|--------|
+| 100,000 | 8 | 256 | 100 | 1000 | leader only | 25,000 | 35 MB |
+| 100,000 | 8 | 256 | 100 | 1000 | all members | 33,000 | 35 MB |
+
+Sample commands are:
+
+```
+benchmark --endpoints={IP_1},{IP_2},{IP_3} --conns=100 --clients=1000 put --key-size=8 --sequential-keys --total=10000 --val-size=256
+benchmark --endpoints={IP_1} --conns=100 --clients=1000 put --key-size=8 --sequential-keys --total=10000 --val-size=256
+```
+
+Linearized-read request gooes through leader for consensus, while serialized-read request can be served by any etcd member. etcd can read:
+
+| Number of requests | Key size in bytes | Value size in bytes | Number of connections | Number of clients | Consistency | Average read QPS |
+|--------------------|-------------------|---------------------|-----------------------|-------------------|-------------|------------------|
+| 100,000 | 8 | 256 | 100 | 1000 | Linearizable | 43,000 |
+| 100,000 | 8 | 256 | 100 | 1000 | Serializable | 93,000 |
+
+Sample commands are:
+
+```
+benchmark --endpoints={IP_1},{IP_2},{IP_3} --conns=100 --clients=1000 range YOUR_KEY --consistency=l --total=100000
+benchmark --endpoints={IP_1} --conns=100 --clients=1000 range YOUR_KEY --consistency=s --total=100000
+```
+
+We encourage to run the benchmark test on your own cluster. etcd performance might vary from different environments.
```

Comment:
We encourage running the benchmark test when setting up an etcd cluster for the first time in a new environment to ensure the cluster achieves adequate performance; cluster latency and throughput can be sensitive to minor environment differences.